src/app.md

```diff
@@ -632,6 +633,54 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-semver-complex]: #ex-semver-complex
+<a name="ex-semver-complex"></a>
+## Parse a complex version string.
+
+[![semver-badge]][semver] [![cat-config-badge]][cat-config]
+
+Constructs a [`semver::Version`] from a complex version string using [`Version::parse`]. The string
+contains pre-release and build metadata as defined in the [Semantic Versioning Specification].
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate semver;
+
+use semver::{Identifier, Version};
+#
+# error_chain! {
+#     foreign_links {
+#         SemVer(semver::SemVerError);
+#     }
+# }
+
+fn run() -> Result<()> {
+    let version_str = "1.0.49-125+g72ee7853";
+    let version = Version::parse(version_str)?;
+
+    assert_eq!(version.major, 1);
+    assert_eq!(version.minor, 0);
+    assert_eq!(version.patch, 49);
+
+    assert_eq!(version.pre.len(), 1);
+    assert_eq!(version.pre[0], Identifier::Numeric(125));
+
+    assert_eq!(version.build.len(), 1);
+    assert_eq!(
+        version.build[0],
+        Identifier::AlphaNumeric(String::from("g72ee7853"))
+    );
+
+    let version_string = version.to_string();
```

Comment:
i would rename the variable to `serialized_version` or something along these lines to emphasize what is going on here.